students/MzKhan/lesson03/mailroom.py

```diff
@@ -0,0 +1,153 @@
+              ['Kimberley', 'Allen', 12000.99, 5000],
+              ['Doug', 'Boolinger', 260.57, 930.90],
+              ['Sherry', 'Henning', 6000, 2460.20, 900] ]
+
+def prompt_user():
+    # This method prompts the user for the valid input.
+    # input: string
+    # return: int
+    print('Please choose one of the following options: \n')
+    print("1: Send a Thank You")
+    print("2: Create a report")
+    print("3: Quit")
+    return int(input("Option:  "))
+
+def thank_you(donors_data = donors_list):
+    # This method asks for the donor's name and the donation amount
+    # from the user. Then, it adds the donation amount for the donor.
+    # It sends the thank you email to the donor for his/her donation.
+    # parm: list    (kwarg)
+
+    name = name_prompt().split(' ')
+    amount = donation_prompt(name)
+    add_donation(name, amount, donors_data)
+    email(name, amount)
+
+def add_donation(name, amount, donors_data = donors_list):
+    # This method adds the donation amount for the given donor.
+    # If the donor is already present in the history, it appends the donation
+    # amount for that user.
+    # If the donor is a new donor, it appends the new donor to the donor's list
+    # parm: list
+    # parm: float
+    # parm: list    (kwarg)
+    donorIndex = find_donor(name,donors_data)
+    if donorIndex < len(donors_data):
+        donors_data[donorIndex].append(amount)
+    else:
+        donors_data.append([name[0],name[1],amount])
+
+def name_prompt():
+    # This method prompts the user for the donor's name OR
+    # check the existing donors.
+    # parm: none
+    # return: string
+    donor_name = input("Please enter the donor's full name OR \n"
+                 +"type 'list' for the existing donors: ")
+    if donor_name == 'list':
+        display_list()
+        donor_name = input("Please enter the donor's full name: ")
+    return donor_name
+
+def donation_prompt(name):
+    # This method prompts the user for the donation amount for the given donor.
+    # parm: list
+    # return: float
+    donation_amount = input("Please enter the donation amount for {} {}: "
+                            .format(*name))
+    return float(donation_amount)
+
+def display_list(donors_data=donors_list):
+    # This method prints all the donor's list for the user.
+    # parm: list    (kwarg)
+    print("\nOur generous donors: \n")
+    for donor in donors_data:
+        print(donor[0],donor[1])
+
+def find_donor(donor_name, donors_data = donors_list):
+    # This method checks if the given donor exists in the donor's list.
+    # If the donor does exist, it returns its index
+    # If the donor doesn't exist, it returns the index to add the new donor.
+    # parm: list
+    # parm: list    (kwarg)
+    # return: int
+    index = len(donors_data)
+    for donor in donors_data:
+        if (donor_name[0] == donor[0] and donor_name[1] == donor[1]):
+            index = donors_data.index(donor)
```

Comment:
if you're looking for something in the list and found you should exit at this point to avoid continuing looping.
so here you can add `return index`

hint: also remember that for loop supports `else` statement that you could utilize here as well